src/EioDriver.php

```diff
@@ -436,6 +306,21 @@ public function mkdir(string $path, int $mode = 0777, bool $recursive = false):
         return $deferred->promise();
     }
 
+    private function isdir(string $path): Promise
+    {
+        $deferred = new Deferred;
+
+        $this->stat($path)->onResolve(function ($error, $result) use ($deferred): void {
+            if ($result) {
+                $deferred->resolve(!($result["mode"] & \EIO_S_IFREG));
```

Comment:
```suggestion
                $deferred->resolve(($result["mode"] & \EIO_S_IFDIR) === \EIO_S_IFDIR);
```

Reply:
@kelunik The constant `EIO_S_IFDIR` does not exist: https://www.php.net/manual/en/eio.constants.php

Reply:
I guess that's something we should add then? https://github.com/rosmanov/pecl-eio/blob/57e1f529c2950ed41d46b559c72a37ca1a46ee97/php7/php_eio.c#L1071

Reply:
I don't think so. eio_mknod (which these constants are meant for) creates a normal file (https://www.php.net/manual/en/function.eio-mknod.php), for directory there is a https://www.php.net/manual/en/function.eio-mkdir.php